disallow irrelevant pages by defaults in robots

bbodenmiller opened the pull request:
Update default robots.txt rules to disallow irrelevant pages that search engines should not care about. This will still allow important pages like the files, commit details, merge requests, issues, comments, etc. to be crawled.

Fixes http://feedback.gitlab.com/forums/176466-general/suggestions/7665033-create-robots-txt

bbodenmiller commented:
Fixed merge conflicts.

robotwood commented:
+1

stanhu commented on CHANGELOG:
```diff
@@ -58,6 +58,7 @@ v 7.14.0 (unreleased)
   - Fix: User search feature in admin area does not respect filters
   - Set max-width for README, issue and merge request description for easier read on big screens
   - Update Flowdock integration to support new Flowdock API (Boyan Tabakov)
+   - Update default robots.txt rules to disallow crawling of irrelevant pages (Ben Bodenmiller)
```
Just a minor issue: an extra space here.

stanhu commented:
Looks like there is a recent conflict with CHANGELOG that needs to be resolved here. Thanks!

bbodenmiller committed:
disallow irrelevant pages by default in robots

Update default robots.txt rules to disallow irrelevant pages that search
engines should not care about. This will still allow important pages
like the files, commit details, merge requests, issues, comments, etc.
to be crawled.

bbodenmiller commented:
@stanhu fixed both issues.

stanhu commented:
Thanks, @bbodenmiller!

gfairchild commented:
I have a question. http://www.robotstxt.org/robotstxt.html has this blurb:

>  Note also that globbing and regular expression are **not** supported in either the User-agent or Disallow lines. The '\*' in the User-agent field is a special value meaning "any robot". Specifically, you cannot have lines like "User-agent: \*bot\*", "Disallow: /tmp/\*" or "Disallow: \*.gif".

However, the robots.txt file now included with GitLab contains lots of these regex-style Disallow lines:

```
# Project details
Disallow: /*/*.git
Disallow: /*/*/fork/new
Disallow: /*/*/repository/archive*
Disallow: /*/*/activity
Disallow: /*/*/new
Disallow: /*/*/edit
Disallow: /*/*/raw
Disallow: /*/*/blame
Disallow: /*/*/commits/*/*
Disallow: /*/*/commit
Disallow: /*/*/compare
Disallow: /*/*/branches/new
Disallow: /*/*/tags/new
Disallow: /*/*/network
Disallow: /*/*/graphs
Disallow: /*/*/milestones/new
Disallow: /*/*/milestones/*/edit
Disallow: /*/*/issues/new
Disallow: /*/*/issues/*/edit
Disallow: /*/*/merge_requests/new
Disallow: /*/*/merge_requests/*.patch
Disallow: /*/*/merge_requests/*.diff
Disallow: /*/*/merge_requests/*/edit
Disallow: /*/*/merge_requests/*/diffs
Disallow: /*/*/project_members/import
Disallow: /*/*/labels/new
Disallow: /*/*/labels/*/edit
Disallow: /*/*/wikis/*/edit
Disallow: /*/*/snippets/new
Disallow: /*/*/snippets/*/edit
Disallow: /*/*/snippets/*/raw
Disallow: /*/*/deploy_keys
Disallow: /*/*/hooks
Disallow: /*/*/services
Disallow: /*/*/protected_branches
```

What should we do about this?

bbodenmiller commented:
@gfairchild the major search engines support wildcards: https://developers.google.com/webmasters/control-crawl-index/docs/robots_txt#url-matching-based-on-path-values

gfairchild commented:
Awesome, that alleviates my concerns. :)
